/*
 * This file was automatically generated by EvoSuite
 * Sat Dec 10 01:34:11 GMT 2022
 */

package cn.byhieg.iotutorial.bytestreamio;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import cn.byhieg.iotutorial.bytestreamio.ByteArrayInputStreamExample;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ByteArrayInputStreamExample_ESTest extends ByteArrayInputStreamExample_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      ByteArrayInputStreamExample byteArrayInputStreamExample0 = new ByteArrayInputStreamExample();
      byte[] byteArray0 = new byte[4];
      byteArrayInputStreamExample0.readMarkAndReset(byteArray0, (byte) (-52));
      assertArrayEquals(new byte[] {(byte)0, (byte)0, (byte)0, (byte)0}, byteArray0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ByteArrayInputStreamExample byteArrayInputStreamExample0 = new ByteArrayInputStreamExample();
      // Undeclared exception!
      try { 
        byteArrayInputStreamExample0.readMarkAndReset((byte[]) null, 0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.ByteArrayInputStream", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ByteArrayInputStreamExample byteArrayInputStreamExample0 = new ByteArrayInputStreamExample();
      // Undeclared exception!
      try { 
        byteArrayInputStreamExample0.readFromArray((byte[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.ByteArrayInputStream", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ByteArrayInputStreamExample byteArrayInputStreamExample0 = new ByteArrayInputStreamExample();
      byte[] byteArray0 = new byte[17];
      String string0 = byteArrayInputStreamExample0.readFromArray(byteArray0);
      assertEquals("00000000000000000", string0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      ByteArrayInputStreamExample byteArrayInputStreamExample0 = new ByteArrayInputStreamExample();
      byte[] byteArray0 = new byte[0];
      String string0 = byteArrayInputStreamExample0.readFromArray(byteArray0);
      assertEquals("", string0);
  }
}
